File: ios/Authzero.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Authzero, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(logout:
              withResolver:(RCTPromiseResolveBlock)resolve
              withRejecter:(RCTPromiseRejecterBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return YES;
}

@end

File: ios/Authzero.swift
@objc(Authzero)
class Authzero: NSObject {

  @objc(multiply:withB:withResolver:withRejecter:)
  func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
    resolve(a*b)
  }

  @objc(logout:withResolver:)
  func logout(resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
    // do stuff here
      resolve("Hello")
    }
}
